internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -1032,6 +1035,15 @@ func (c *Coordinator) runLoopIteration(ctx context.Context) {
 
 	case c.heartbeatChan <- struct{}{}:
 
+	case <-c.servicePidUpdate:
+		c.logger.Infof("got pid service update, refreshing config")
+		err := c.refreshComponentModel(ctx)
```

Comment:
If endpoint crashloops, which I've seen happen, we are going to be doing this continuously.

I suspect this is expensive and we likely want to rate limit it.

Reply:
This will be particularly bad if multiple sub-processes were to enter crash loops, or just restart frequently.

Reply:
yeah, the debug line isn't supposed to be there, oops. Do you think the overall routine should have a rate limit of some kind? We do a PID comparison in `applyComponentState`, so we won't be hammering the update if we get the same value a bunch of times. 

If we want to handle a sort of crashloop scenario, (endpoint restarts frequently, each time coming up with a new PID) that could be interesting, since we'd still want to track the PID, so we can actually update the config once the crash loop stops or slows down?

Reply:
Hmm, I see, that makes sense. So it sounds like we need some kind of separate thread that operates on a delay or otherwise bundles the PID updates.